internal/xds/translator/testdata/out/xds-ir/credential-injection-backend-filter.routes.yaml

```diff
@@ -0,0 +1,40 @@
+- ignorePortInHostMatching: true
+  name: envoy-gateway/gateway-1/http
+  virtualHosts:
+  - domains:
+    - '*'
+    metadata:
+      filterMetadata:
+        envoy-gateway:
+          resources:
+          - kind: Gateway
+            name: gateway-1
+            namespace: envoy-gateway
+            sectionName: http
+    name: envoy-gateway/gateway-1/http/*
+    routes:
+    - match:
+        pathSeparatedPrefix: /foo
+      metadata:
+        filterMetadata:
+          envoy-gateway:
+            resources:
+            - kind: HTTPRoute
+              name: httproute-1
+              namespace: default
+      name: httproute/default/httproute-1/rule/1/match/0/*
+      route:
+        clusterNotFoundResponseCode: INTERNAL_SERVER_ERROR
+        upgradeConfigs:
+        - upgradeType: websocket
+        weightedClusters:
```

Comment:
this is generating `weightedClusters` logic which is not required
https://github.com/envoyproxy/gateway/blob/6863a17a99eec2a46623fd8b275a2dae186223db/internal/xds/translator/route.go#L724 needs to be updated

Reply:
this is needed, and is working as expected, my bad